src/utils/conversion.ts

```diff
@@ -4365,3 +4368,59 @@ export function metadataValueDetailToMeshMetadataValueDetail(
 export function instructionMemoToString(value: U8aFixed): string {
   return removePadding(hexToString(value.toString()));
 }
+
+/**
+ * @hidden
+ */
+export function middlewareInstructionToHistoricInstruction(
+  instruction: Instruction,
+  context: Context
+): HistoricInstruction {
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+  const {
+    id: instructionId,
+    status,
+    settlementType,
+    endBlock,
+    tradeDate,
+    valueDate,
+    legs: { nodes: legs },
+    memo,
+    createdBlock,
+    venueId,
+  } = instruction;
+  const { blockId, hash, datetime } = createdBlock!;
+
+  let typeDetails;
+
+  if (settlementType === InstructionType.SettleOnBlock) {
+    typeDetails = {
+      type: InstructionType.SettleOnBlock,
+      endBlock: new BigNumber(endBlock!),
+    };
+  } else {
+    typeDetails = {
+      type: InstructionType.SettleOnAffirmation,
+    };
+  }
+
+  return {
+    id: new BigNumber(instructionId),
+    blockNumber: new BigNumber(blockId),
+    blockHash: hash,
+    status: status as InstructionStatusEnum,
```

Comment:
(nit) I think this works. If not OK to leave

```suggestion
    status: status,
```

Reply:
I have updated the same in the next PR